dashboard/lib/dynamic_config/datastore_cache.rb

```diff
@@ -0,0 +1,72 @@
+# A caching layer that sits in front of a datastore that
+# implements get and set
+
+class DatastoreCache
+  # @param datastore [Object] a datastore adapter
+  # @param cache_expiration [int] seconds after which a cached entry expires
+  def initialize(datastore, cache_expiration: 30)
```

Comment:
Some quick envelope math:  I think at 30 seconds we are OK for HOC.  If we had 100 frontends servers with 32 processes each polling every 30 seconds that's only about 100 QPS of polling activity.   (No action required)